src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -728,28 +739,41 @@ WorkerShardStats(char *nodeName, uint32 nodePort, Oid relationId, char *shardNam
 	appendStringInfo(partitionValueQuery, SHARD_RANGE_QUERY,
 					 partitionColumnName, partitionColumnName, shardName);
 
-	queryOK = MultiClientExecute(connectionId, partitionValueQuery->data,
-								 &queryResult, &rowCount, &columnCount);
-	if (!queryOK)
+	querySent = SendRemoteCommand(connection, partitionValueQuery->data);
+	if (!querySent)
+	{
+		ReportConnectionError(connection, WARNING);
+		return false;
+	}
+
+	queryResult = GetRemoteCommandResult(connection, doRaiseInterrupts);
+	if (!IsResponseOK(queryResult))
 	{
-		MultiClientDisconnect(connectionId);
+		ReportResultError(connection, queryResult, WARNING);
+		PQclear(queryResult);
+		ForgetResults(connection);
 		return false;
 	}
```

Comment:
That makes sense eventually. We already have `ExecuteCriticalRemoteCommand`, adding `ExecuteRemoteCommand` sounds reasonable.

 This pattern only occurs three times in the current codebase (including these two, the third is [here](https://github.com/citusdata/citus/blob/master/src/backend/distributed/transaction/transaction_recovery.c#L241)) and it's not clear to me how common it'll eventually be, maybe it's worth waiting?